go/libkb/upak_loader.go

```diff
@@ -417,19 +417,31 @@ func (u *CachedUPAKLoader) loadWithInfo(arg LoadUserArg, info *CachedUserLoadInf
 	if !arg.forceReload {
 		upak, fresh = u.getCachedUPAK(ctx, arg.uid, arg.stubMode, info)
 	}
-	if arg.forcePoll {
-		g.VDL.CLogf(ctx, VLog0, "%s: force-poll required us to repoll (fresh=%v)", culDebug(arg.uid), fresh)
-		fresh = false
-	}
 
 	if upak != nil {
-		g.VDL.CLogf(ctx, VLog0, "%s: cache-hit; fresh=%v", culDebug(arg.uid), fresh)
-		if fresh || arg.staleOK {
-			return returnUPAK(upak, true)
-		}
-		if arg.cachedOnly {
-			return nil, nil, UserNotFoundError{UID: arg.uid, Msg: "cached user found, but it was stale, and cached only"}
+		// Either there is no cached UPAK or forceReload is set.
+
+		// If we are not forced to repoll, we might be done here.
+		if !arg.forcePoll {
+			if fresh {
+				// Cache hit, not stale.
+				return returnUPAK(upak, true)
+			}
+			if arg.staleOK {
+				// Stale cache allowed.
+				return returnUPAK(upak, true)
+			}
+			if arg.cachedOnly {
+				// Stale cache not allowed but cache was stale.
+				return nil, nil, UserNotFoundError{UID: arg.uid, Msg: "cached user found, but it was stale, and cached only"}
+			}
```

Comment:
what about this else case?

Reply:
In general, I'm not a fan of nest if statements. From [here](https://golang.org/doc/effective_go.html#if), there's this:


> This is an example of a common situation where code must guard against a sequence of error conditions. The code reads well if the successful flow of control runs down the page, eliminating error cases as they arise. Since error cases tend to end in return statements, the resulting code needs no else statements.


Can we make this flow more "linear."  Maybe but cutting out into a subfunction or with a series of linear non-nested if statements?  This function is already so tricky, I don't want it to become harder to read.

Reply:
I stared at this for a while and didn't think of any way to make it clearer; Patrick suggested the nested ifs and I think I agree with him that it's clearer than the alternative. WDYT?